### ISSUE TITLE
Enslaved Farmers aren't farming

### ISSUE THREAD

**Issue by KA101** [SoftMonster/KeeperRLCommunityResources#16]
https://github.com/SoftMonster/KeeperRLCommunityResources/blob/880b773f79df698e357bf3b19d8cf86bdacb4ab9/Mods/Alpha36/A36BonusMod/furniture.txt#LL50C5-L50C5

I'm guessing these folks are white_knight's workforce, and that captured peasants enslaved by an Evil keeper don't qualify.

Asking because my slave peasants go out and stand around on the crop field, but don't seem to be generating any crops (4 peasants over 4-5 days of game time or so).  While it's important to not have "Villain" Peasants just piling crops up everywhere, evil keepers would reasonably expect "farmer" units to farm.

(Check: I converted one of the peasants, took control, and headed out to farm.  No option to scythe the field.  Imps don't seem to have "Farmer", despite the code looking for them to scythe.)

**Comment by KA101** [comment]
I looked into giving Imps back farming capability (would resolve the underlying problem of evil keepers being unable to farm grain) and it looks like you removed that as part of a compatibility fix.
https://github.com/KA101/KeeperRLCommunityResources/blob/2f8adf6d01ef9ed5e68ea4d5567a3e29a2d836b5/Mods/Alpha36/A36BonusMod/creatures.txt#L1993 , currently dig-speed 30, used to be crops_skill.

I'm guessing, without being sure, that crops_skill was breaking the Farming mod?  Wasn't sure but thought it better to check.

**Comment by SoftMonster** [comment]
Friendly villages shouldn’t produce grain

**Comment by SoftMonster** [comment]
Needs some extra thought

**Comment by KA101** [comment]
> Needs some extra thought

Fair.  I'm not sure whether Farming was fully-integrated into BonusMod or if there's some reason to use them separately.  If integrated, sure, Imps should be able to farm.  If not, then not...but players need to know because then we've got a 25 grain starting credit, and not much we can do with it w/o Farming.  (Or making it tangible and reinstating eating it.)

Thanks.

**Comment by KA101** [comment]
(And just for the record: yes, these were crops I'd ordered and the workforce planted.  Wasn't a preexisting crop field.)

**Comment by KA101** [comment]
Looked some more and *might* be a mod conflict with DungeonDesign, since the cosmetic crops (build with wood) are also registering as present.  Not sure, will mention on discord.

(I'd suspect the conflict, if it exists, would mean that wooden crops would suffice for farmers as well, so an exploit rather than a bug. ?? )

**Comment by KA101** [comment]
OK, I can rule out the mod conflict.  Just fired up a Less-Evil BonusMod only run, with the relevant chances cranked up to 1.0.
https://github.com/SoftMonster/KeeperRLCommunityResources/blob/3ee54e649969ea2b36e21277f8659e53a97acb92/Mods/Alpha36/A36BonusMod/furniture.txt#LL47C4-L47C5

Still not getting any sacks, but there's another bug that's a confounding factor:
The white_knight keeper spends as much of its starting CROPS credit as it can on recruiting peasants, rather than bringing 4 and keeping the full amount.
(25 crops? you get 6 peasants and 4 crops.  Doubled it to 50, got 7 peasants and 16 crops.  #ThatJustAin'tRight )

**Comment by KA101** [comment]
With 16 crops we got 8 tiles of farm, so watched the peasants scythe the field for a few hundred turns.  (Remember, I have the rates cranked to 1.0 for scything and for crop sacks to drop when peasants scythe.)

I've got to get some sleep.  If I had to guess, the chain is rolling the percentages in order and any failure stops the process, but I've been at it a while and was hoping to get to bed an hour or so ago.
